prow/crier/controller.go

```diff
@@ -166,27 +194,12 @@ func (r *reconciler) reconcile(log *logrus.Entry, req reconcile.Request) (*recon
 		return requeue, nil
 	}
 
-	log.Info("Reported job, now will update pj")
+	log.Info("Reported job(s), now will update pj(s)")
 	for _, pjob := range pjs {
```

Comment:
Please move this for loop out of this function. This is the second time reusing a stale `pj` variable has caused an outage (previously this was from https://github.com/kubernetes/test-infra/pull/18065)

Reply:
I moved the loop contents into its own function to help with that. I figured putting the loop itself in the helper function would increase the chance of reusing a stale PJ rather than reduce it since multiple PJs would be accessible from the function.

WDYT? I'm happy to move the loop into the function if that is still preferable.

Reply:
I'd prefer to put it in its own function but can do that later. This is much better. Thanks.